usbtmc/usbtmc.py

```diff
@@ -507,46 +497,44 @@ def read_raw(self, num=-1):
         while not eom:
             if not self.rigol_quirk or read_data == b'':
 
-                # if the rigol sees this again, it will restart the transfer
+                # if the Rigol sees this again, it will restart the transfer
                 # so only send it the first time
 
-                req = self.pack_dev_dep_msg_in_header(read_len, term_char)
+                req = self._pack_dev_dep_msg_in_header(read_len, term_char)
                 self.bulk_out_ep.write(req)
             
-            resp = self.bulk_in_ep.read(read_len+USBTMC_HEADER_SIZE+3, timeout = int(self.timeout*1000))
+            resp = self.bulk_in_ep.read(read_len+USBTMC_HEADER_SIZE+3, timeout=int(self.timeout*1000))
 
             if sys.version_info >= (3, 0):
                 resp = resp.tobytes()
             else:
                 resp = resp.tostring()
             
-            if self.rigol_quirk and read_data:
-                pass # do nothing, the packet has no header if it isn't the first
-            else:
-                msgid, btag, btaginverse, transfer_size, transfer_attributes, data = self.unpack_dev_dep_resp_header(resp) 
-
-
             if self.rigol_quirk:
-                # rigol devices only send the header in the first packet, and they lie about whether the transaction is complete
+                # Rigol devices only send the header in the first packet,
+                # and they lie about whether the transaction is complete
                 if read_data:
                     read_data += resp
```

Comment:
If the condition on the old line 523 was hit, this would never happen. Perhaps instead of `read_data += resp`, this should be `pass`, as it's the same condition as in the old line 523.